src/services/download/http.rs

```diff
@@ -79,49 +77,48 @@ pub fn download_source(
                 builder.finish()
             }),
         )
+        .boxed_local()
+        .compat()
     });
-
     let response = Retry::spawn(
         ExponentialBackoff::from_millis(10).map(jitter).take(3),
-        response,
-    );
-
-    let response = response.map_err(|e| match e {
+        connect,
+    )
+    .compat()
+    .await
+    .map_err(|e| match e {
         tokio_retry::Error::OperationError(e) => e,
         e => panic!("{}", e),
     });
```

Comment:
Suggestions on how retry should be done done in an async-fn world are appreciated.  This isn't particularly nice, but I stayed with it because so far it seems we'd have to keep building on `tokio::Delay` which is futures01 for us.  I couldn't find anything that replaced the tokio-retry crate however.  Maybe I was looking for the wrong thing.